src/main/java/io/vertx/serviceproxy/HelperUtils.java

```diff
@@ -108,5 +100,21 @@
     return new HashSet<T>((List<T>)list);
   }
 
+  public static JsonObject generateDebugInfo(Throwable cause) {
+    if (cause == null) return null;
+    JsonObject obj = new JsonObject();
+    obj.put("causeName", cause.getClass().getCanonicalName());
```

Comment:
What if the exception has [suppressed exceptions](https://docs.oracle.com/javase/7/docs/api/java/lang/Throwable.html#getSuppressed())?

Reply:
We can serialize them to in a field called "suppressed"